pkg/hydrate/tool_util.go

```diff
@@ -326,7 +326,11 @@ func ValidateOptions(ctx context.Context, rootDir cmpath.Absolute, apiServerTime
 }
 
 func newClientClient(cfg *rest.Config) (client.Client, error) {
-	mapper, err := apiutil.NewDynamicRESTMapper(cfg)
+	httpClient, err := rest.HTTPClientFor(cfg)
```

Comment:
Is there a way to pass this httpclient through to the client.New so it uses the same one? Or does the client.Client not use a rest.Client?

Reply:
They both use `http.Client`. Looks like it can be passed into [client.Options](https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/vendor/sigs.k8s.io/controller-runtime/pkg/client/client.go#L42) when invoking `client.New`. If not provided through options, it [calls the same function](https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/vendor/sigs.k8s.io/controller-runtime/pkg/client/client.go#L138) to instantiate a new `http.Client`.

So we could probably pass this in to avoid duplicate instantiation